src/tests/unique_path_unittest.cc

```diff
@@ -0,0 +1,295 @@
+// in the documentation and/or other materials provided with the
+// distribution.
+//     * Neither the name of Google Inc. nor the names of its
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+// ---
+// Author: Artem Y. Polyakov
+
+#include <stdlib.h>   // for environment primitives
+#include <unistd.h>   // for getpid()
+#include <limits.h>   // for PATH_MAX
+#include <string>
+#include <utility>
+#include "addressmap-inl.h"
+#include "base/logging.h"
+#include "base/sysinfo.h"
+#include "base/commandlineflags.h"
+
+#define TEST_VAR   "GPROF_TEST_PATH"
+#define TEST_FORCE TEST_VAR "_USE_PID"
+#define TEST_VAL   "/var/log/some_file_name"
+#define HPC_RANK "5"
+
+#define PID_SUFFIX ".PID-"
+#define PMIX_SUFFIX ".PMIxRank-"
+#define SLURM_SUFFIX ".SlurmID-"
+
+// Manage environment
+void setEnvDefault() {
+  setenv(TEST_VAR, TEST_VAL, 1);
+}
+
+void unsetEnvDefault() {
+  unsetenv(TEST_VAR);
+}
+
+void setEnvForced() {
+  setEnvDefault();
+  setenv(TEST_FORCE, "1", 1);
+}
+
+void unsetEnvForced() {
+  unsetEnvDefault();
+  unsetenv(TEST_FORCE);
+}
+
+// Possible outcomes
+
+void appendPID(std::string &str)
+{
+  str += PID_SUFFIX;
+  str += std::to_string(getpid());
+}
+
+const std::string genDefaultParent()
+{
+  std::string expected;
+  expected = TEST_VAL;
+  return expected;
+}
+
+const std::string genDefaultChild()
+{
+  std::string expected = genDefaultParent();
+  appendPID(expected);
+  return expected;
+}
+
+const std::string genForced()
+{
+  return genDefaultChild();
+}
+
+const std::string genPMIxParent()
+{
+  std::string expected = genDefaultParent();
+  expected += PMIX_SUFFIX;
+  expected += HPC_RANK;
+  return expected;
+}
+
+const std::string genPMIxChild()
+{
+  std::string expected = genPMIxParent();
+  appendPID(expected);
+  return expected;
+}
+
+const std::string genSlurmParent()
+{
+  std::string expected = genDefaultParent();
+  expected += SLURM_SUFFIX;
+  expected += HPC_RANK;
+  return expected;
+}
+
+const std::string genSlurmChild()
+{
+  std::string expected = genSlurmParent();
+  appendPID(expected);
```

Comment:
@alk here for example, the child process in Slurm environment would have both Slurm and PID suffixes.

I decided to go ahead and always append to make sure that this is easily extendable, i.e. you can decide to add more suffixes when needed without breaking the logic.